### ISSUE TITLE
error routing on alpha 34

### ISSUE THREAD

**Issue by monmonja** [angular/angular#3556]
im getting router_1.RouteConfig is not a function on 34 but when i switch back to 33 its okay. Is there a change in name?

**Comment by pkozlowski-opensource** [comment]
`RouteConfig` is still there (https://code.angularjs.org/2.0.0-alpha.34/router.dev.js). We need more info: what is the setup you are using (transpilation, module loader etc.)? What is exact error (full stack trace)? 

The best is to share a running reproduce scenario in a plunker, you might want to start with this one: http://plnkr.co/edit/wQzOPJXrhPCNznQc3UYB?p=preview

**Comment by Zyzle** [comment]
I'm seeing the same issue in my application after the upgrade to alpha 34 but I've not seen any mention of breaking changes with regards to the router in this release.  Something missing from the changelog maybe?

My routeConfig looks like this:

```typescript
@RouteConfig([
  { path: "/", as: "home", component: HomePage },
  { path: "/formtest", as: "formtest", component: FormTest },
  { path: "/ghrepos", as: "ghrepos", component: GhRepos },
  { path: "/snippets", as: "snippets", component: SnippetsList }
])
```

I'm using traceur and systemjs thusly:

```html
<script src="https://github.jspm.io/jmcriffey/bower-traceur-runtime@0.0.87/traceur-runtime.js"></script>
<script src="https://jspm.io/system@0.16.js" charset="utf-8"></script>
```

I'll try and add a more complete stack trace later

**Comment by pkozlowski-opensource** [comment]
It might be due to https://github.com/angular/angular/issues/3528 but impossible to say without more detailed example and / or stack trace.

**Comment by Zyzle** [comment]
Sorry I'm away from my main terminal right now, I'll try and build here and get a more complete stack trace.  I would also say I built the angular/router js files and typings from the alpha34 tag rather than using those from code.angularjs.org/Definitelytyped but thats never been an issue in the past

**Comment by TiagoBraga39** [comment]
@Zyzle Shouldn't you have "new Route" on your routes? In alpha 33, it would be an error: https://github.com/angular/angular/blob/fdffcaba9b609651742176650b300dcc5a8baf42/modules/angular2/test/router/router_integration_spec.ts#L185

But anyway, the error still persists for me.

**Comment by monmonja** [comment]
I have the same config as @Zyzle , Here is my trace

<pre>TypeError: Error loading "main" at http://localhost:8080/js/main.js
router_1.RouteConfig is not a function
router_1.RouteConfig is not a function
    at http://localhost:8080/js/app.js:25:22
    at Function.<anonymous> (http://localhost:8080/js/app.js:30:7)
    at $__global.upgradeSystemLoader.c.execute (https://jspm.io/system@0.16.11.js:5:14731)
    at f (https://jspm.io/system@0.16.11.js:5:5970)
    at f (https://jspm.io/system@0.16.11.js:5:5932)
    at l (https://jspm.io/system@0.16.11.js:5:4597)
    at Object.$__global.upgradeSystemLoader.e.instantiate.Promise.all.then.execute (https://jspm.io/system@0.16.11.js:5:8278)
    at h (https://jspm.io/es6-module-loader@0.16.6.js:1:19111)
    at m (https://jspm.io/es6-module-loader@0.16.6.js:1:19483)
    at l (https://jspm.io/es6-module-loader@0.16.6.js:1:16727)</pre>

**Comment by Zyzle** [comment]
@TiagoBraga39 I see the example in [inbox-app](https://github.com/angular/angular/blob/master/modules/examples/src/routing/inbox-app.ts#L144) uses this format however the documentation in [router-link](https://github.com/angular/angular/blob/c0adae69ac33062b2668b1d87a8dcefba28127fb/modules/angular2/src/router/router_link.ts#L13) uses the same layout as my example (and it did still work in alpha 33, I believe)

**Comment by TiagoBraga39** [comment]
@Zyzle I can confirm your example also works. Perhaps those new Route and new Redirect are not finished, I don't know why should they both work.

**Comment by TiagoBraga39** [comment]
@Zyzle I remembered something. I guess RouteConfig changed its location. My imports:
```javascript
import { routerInjectables, routerDirectives, Router, RouterLink, RouterOutlet, LocationStrategy, HashLocationStrategy } from 'angular2/router';
import { RouteConfig, Route, AsyncRoute, Redirect } from 'angular2/src/router/route_config_decorator';
```

**Comment by theFifthElf** [comment]
I have the same issue (and also the same problem trying to use `new Route(...) `

Uncaught TypeError: router_1.RouteConfig is not a function
(anonymous function) @ portal_app.js:36
(anonymous function) @ portal_app.js:43
(anonymous function) @ portal_app.js:46
doEval @ system.src.js:2167
__eval @ system.src.js:2138exec @ system.src.js:211
$__global.upgradeSystemLoader.loader.instantiate.load.metadata.execute @ ```

**Comment by markharding** [comment]
I am also getting this error in alpha 34

```
Uncaught TypeError: router_1.RouteConfig is not a function
```

**Comment by pkozlowski-opensource** [comment]
@markharding / All - could one of you guys put together a plunker that reproduces the issue? ou can start with this one: http://plnkr.co/edit/wH92E7Vr0RMk69gQ6WVx?p=preview (it contains work-around for   #3528)

**Comment by Zyzle** [comment]
@pkozlowski-opensource Sorry this took so long, here http://plnkr.co/edit/T2MxnEdRlCqkfU1jNqVm  This should be the simplest case that will reproduce the error, I can't seem to make the plunkr work correctly though (its complaining about base.attr and the instantiation of the Location strategy

**Comment by pkozlowski-opensource** [comment]
@Zyzle OK, so based on your plunker it should be fixed in master, right? I'm going to close this one then as fixed by 1562bc91bae75bd0b8ac58114ba56908400ac99b.

All: if anyone is till seeing this pb, please provide a plunker (you can start with http://plnkr.co/edit/T2MxnEdRlCqkfU1jNqVm?p=preview) - will happily re-open.

**Comment by vascojm82** [comment]
I still keep getting the same error: **zone.js:392 Error: (SystemJS) TypeError: router_deprecated_1.RouteConfig is not a function**
Don't know what is wrong.  Please re-open, here's the Plunker where the problem can be seen:
http://plnkr.co/edit/XyOEY12yK1mHW1uTQZbn?p=preview
